projects/packages/mcp/src/AbilitiesRegistry/Executors/Post/PostGetExecutor.php

```diff
@@ -0,0 +1,254 @@
+ * Only loaded when the ability is actually executed.
+ */
+class PostGetExecutor implements ExecutorInterface {
+
+	/**
+	 * Execute the post get ability.
+	 *
+	 * @param array $input The input parameters.
+	 *
+	 * @return array The post data or error response.
+	 */
+	public function execute( array $input = array() ): array {
+		$original_blog_id = get_current_blog_id();
+		$switched         = false;
+
+		try {
+			// Validate input.
+			if ( empty( $input['post_id'] ) && empty( $input['post_url'] ) ) {
+				return array(
+					'success' => false,
+					'error'   => array(
+						'code'    => 'missing_post_identifier',
+						'message' => 'Either post_id or post_url must be provided',
+						'status'  => 400,
+					),
+				);
+			}
+
+			// Handle site switching if wpcom_site is provided.
+			if ( ! empty( $input['wpcom_site'] ) ) {
+				$target_site    = $input['wpcom_site'];
+				$target_blog_id = null;
+
+				// Determine if it's a blog ID or URL.
+				if ( ctype_digit( (string) $target_site ) ) {
+					$target_blog_id = (int) $target_site;
+
+					// Validate blog ID.
+					if ( $target_blog_id < 1 ) {
+						return array(
+							'success' => false,
+							'error'   => array(
+								'code'    => 'invalid_site_id',
+								'message' => 'Invalid site ID provided',
+								'status'  => 400,
+							),
+						);
+					}
+
+					$blog_details = get_blog_details( $target_blog_id );
+				} else {
+					// It's a URL - decode and get blog details.
+					$target_site = urldecode( $target_site );
+					$target_site = str_replace( '::', '/', $target_site );
+
+					$blog_details = apply_filters( 'jetpack_mcp_get_blog_details_for_url', null, $target_site );
+					if ( ! $blog_details ) {
+						return array(
+							'success' => false,
+							'error'   => array(
+								'code'    => 'site_not_found',
+								'message' => 'Site not found or inaccessible',
+								'status'  => 404,
+							),
+						);
+					}
+				}
+
+				// Validate blog details.
+				if ( ! $blog_details || is_wp_error( $blog_details ) ) {
+					return array(
+						'success' => false,
+						'error'   => array(
+							'code'    => 'site_not_found',
+							'message' => 'Site not found or inaccessible',
+							'status'  => 404,
+						),
+					);
+				}
+
+				$target_blog_id = (int) $blog_details->blog_id;
+
+				// Only switch if it's a different site.
+				if ( $target_blog_id !== $original_blog_id ) {
+					// Check if site is restricted.
+					$is_suspended = apply_filters( 'jetpack_mcp_is_site_suspended', false, $target_blog_id );
+					if ( $is_suspended ) {
+						return array(
+							'success' => false,
+							'error'   => array(
+								'code'    => 'site_suspended',
+								'message' => 'This site has been suspended',
+								'status'  => 403,
+							),
+						);
+					}
+
+					// Check if site is confidential.
+					$is_ai_access_allowed = apply_filters( 'jetpack_site_ai_access_allowed', false, $target_blog_id );
```

Comment:
We also have `jetpack_mcp_site_ai_access_allowed`. Should we unify this into a single filter?

Reply:
We should only be using `jetpack_mcp_site_ai_access_allowed` here. I'll update.